test/orange/orangeinp/IntersectRegion.test.cc

```diff
@@ -1999,6 +1999,155 @@ TEST_F(PrismTest, rhex)
     EXPECT_VEC_SOFT_EQ((Real3{1, 1.1547005383793, 2}), result.exterior.upper());
 }
 
+//---------------------------------------------------------------------------//
+// REVOLVEDSPECIALTRAPEZOID
+//---------------------------------------------------------------------------//
+using RevolvedSpecialTrapezoidTest = IntersectRegionTest;
+
+//---------------------------------------------------------------------------//
+/*
+ * Test revolving a square around z.
+ */
+TEST_F(RevolvedSpecialTrapezoidTest, simple_cube)
+{
+    SpecialTrapezoid trap({{1, 2}, 1}, {{1, 2}, 2});
+    auto result = this->test(RevolvedSpecialTrapezoid(std::move(trap)));
+
+    static char const expected_node[] = "all(+0, -1, +2, -3)";
+    static char const* const expected_surfaces[]
+        = {"Plane: z=1", "Plane: z=2", "Cyl z: r=1", "Cyl z: r=2"};
+
+    EXPECT_EQ(expected_node, result.node);
+    EXPECT_VEC_EQ(expected_surfaces, result.surfaces);
+
+    EXPECT_VEC_SOFT_EQ((Real3{-2, -2, 1}), result.exterior.lower());
+    EXPECT_VEC_SOFT_EQ((Real3{2, 2, 2}), result.exterior.upper());
+}
+
+//---------------------------------------------------------------------------//
+/*
+ * Test revolving a square around z, but now one segment is coincident
+ * with the z axis, so we don't get an interior cylinder
+ */
+TEST_F(RevolvedSpecialTrapezoidTest, coincident_segment)
+{
+    SpecialTrapezoid trap({{0, 2}, 1}, {{0, 2}, 2});
+    auto result = this->test(RevolvedSpecialTrapezoid(std::move(trap)));
+
+    static char const expected_node[] = "all(+0, -1, -2)";
+    static char const* const expected_surfaces[]
+        = {"Plane: z=1", "Plane: z=2", "Cyl z: r=2"};
+
+    EXPECT_EQ(expected_node, result.node);
+    EXPECT_VEC_EQ(expected_surfaces, result.surfaces);
+
+    EXPECT_VEC_SOFT_EQ((Real3{-2, -2, 1}), result.exterior.lower());
+    EXPECT_VEC_SOFT_EQ((Real3{2, 2, 2}), result.exterior.upper());
+}
+
+//---------------------------------------------------------------------------//
+/*
+ * Test a pointy bottom triangle.
+ *
+ * \verbatim
+   3 _|__________________________
+      |\                        /
+   2 _| \                    /
+      |  \               /
+   1 _|   \          /
+      |    \     /
+   0 _|_____\/_____________________
+      |      |      |      |      |
+      0      1      2      3      4
+   \endverbatim
+ */
+TEST_F(RevolvedSpecialTrapezoidTest, pointy_bot)
+{
+    SpecialTrapezoid trap({{1, 1}, 0}, {{0, 4}, 3});
+    auto result = this->test(RevolvedSpecialTrapezoid(std::move(trap)));
+
+    static char const expected_node[] = "all(+0, -1, +2, -3)";
+    static char const* const expected_surfaces[] = {"Plane: z=0",
+                                                    "Plane: z=3",
+                                                    "Cone z: t=0.33333 at "
+                                                    "{0,0,3}",
+                                                    "Cone z: t=1 at {0,0,-1}"};
```

Comment:
You might get some complains from clang about odd linebreak; add a trailing comma before the end brace and re-join the broken string:
```suggestion
    static char const* const expected_surfaces[] = {
        "Plane: z=0",
        "Plane: z=3",
        "Cone z: t=0.33333 at {0,0,3}",
        "Cone z: t=1 at {0,0,-1}",
    };
```

Reply:
My clang-format automatically does this line break, not me!